src/main.rs

```diff
@@ -137,12 +156,22 @@ impl State {
             .expect("Unable to create surface");
 
         // Create the EGL surface.
-        let config = context.config_id();
-        let native_surface = WlEglSurface::new(surface.id(), self.size.width, self.size.height)
-            .expect("Unable to create EGL surface");
-        let pixel_format = context.pixel_format().expect("No valid pixel format present");
-        let egl_surface = EGLSurface::new(&display, pixel_format, config, native_surface, None)
-            .expect("Unable to bind EGL surface");
+        let mut raw_window_handle = WaylandWindowHandle::empty();
+        raw_window_handle.surface = surface.id().as_ptr().cast();
+        let raw_window_handle = RawWindowHandle::Wayland(raw_window_handle);
+        let surface_attributes = SurfaceAttributesBuilder::<WindowSurface>::new().build(
+            raw_window_handle,
+            NonZeroU32::new(self.size.width as u32).unwrap(),
+            NonZeroU32::new(self.size.height as u32).unwrap(),
```

Comment:
The NonZero stuff can be quite inconvenient at times, but I think it's a great choice here considering I had to figure out why the hell my stuff wasn't working when creating a 0x0 egl surface at first.

Reply:
Yeah, that's intention.